TourMate/TourMate/Frontend/Plan/Views/PlansViews/Components/PlanCardView.swift

```diff
@@ -7,29 +7,38 @@
 
 import SwiftUI
 
+@MainActor
 struct PlanCardView: View {
 
-    @StateObject var viewModel: PlanViewModel
     let planUpvoteViewModel: PlanUpvoteViewModel
+    let planName: String
+    let planStatus: PlanStatus
+    let startDateTime: DateTime
+    let endDateTime: DateTime
     let date: Date
 
-    init(viewModel: PlanViewModel, date: Date) {
-        self._viewModel = StateObject(wrappedValue: viewModel)
-        self.planUpvoteViewModel = ViewModelFactory.getPlanUpvoteViewModel(planViewModel: viewModel)
+    init(planUpvoteViewModel: PlanUpvoteViewModel, planName: String,
+         planStatus: PlanStatus, startDateTime: DateTime, endDateTime: DateTime,
+         date: Date) {
+        self.planUpvoteViewModel = planUpvoteViewModel
+        self.planName = planName
+        self.planStatus = planStatus
+        self.startDateTime = startDateTime
+        self.endDateTime = endDateTime
         self.date = date
     }
```

Comment:
Would it be better if it is just this?
```suggestion
    init(planUpvoteViewModel: PlanUpvoteViewModel, plan: Plan) {
        self.planUpvoteViewModel = planUpvoteViewModel
        self.plan = plan
    }
```
---
Then if we want to access any of the `Plan` properties we can for example, 
```
Text(plan.name).font(.headline)
```

Reply:
yeah i think its better as well, but prof seems to dislike passing the data object in

Reply:
anyway i will change it